include/deal.II/base/symmetric_tensor.h

```diff
@@ -406,6 +411,11 @@ namespace internal
        */
       reference operator [] (const unsigned int);
 
+      /**
+       * Index operator.
+       */
+      reference operator [] (const unsigned int) const;
```

Comment:
I though about that as well. I think it's probably fine -- the `const` at the end refers to the enclosing object, not the tensor itself.